### PR TITLE
fix(team): give the create-mode lead the authoritative roster

### PR DESCRIPTION
**Pull request by NeatCode-Labs**
## Problem

Create a team with, say, `local-worker` and `cursor-worker`, and give it a first prompt like *"create one task per teammate: each writes hello-&lt;their-name&gt;.md"*. The lead frequently creates tasks for teammates that do not exist — `alice`, `bob`, `engineer-1` — and then writes the files itself instead of waiting for the real teammates.

The root cause is an asymmetry between how the lead is briefed on the two entry paths:

* **Launch / resume hydration** goes through `buildPersistentLeadContext`, which ends with a `Members:` block built from `buildMembersPrompt(members)`. The lead sees the real roster.
* **Create mode** defers the user's first prompt and delivers it separately, at the very start of `runDeterministicCreateSpawnFlow`:

  ```ts
  const initialUserPrompt = request.prompt?.trim() ?? '';
  ```

  That bare string is everything the lead has at that moment. This line is unchanged since 785c16d21 (`refactor(team): reapply team provisioning round2`).

The timing makes it worse: the deferred create prompt is delivered **before any teammate has confirmed bootstrap**. So the lead is asked to delegate at the exact moment when (a) it has no teammate names in context and (b) nobody is available to delegate to. Both failure modes follow directly — the model fills the gap with plausible names, and, seeing no available owners, does the work itself.

Because the invented names are then written into task owners and `SendMessage` targets, the damage outlives the first turn: tasks are owned by nobody, DMs go nowhere, and the real teammates sit idle once they do join.

## Fix

* **The deferred create prompt now carries the roster.** `buildCreateBootstrapUserPrompt(prompt, members)` wraps the user's text with an authoritative roster block (from the team configuration, lead excluded, workflow bodies stripped to keep it compact) followed by short integrity rules:
  * teammates are **exactly** these names, no other teammate exists;
  * never invent, guess, rename or paraphrase teammate names — use the exact names in task owners, `SendMessage` `to`, and agent spawns;
  * "one per teammate" means exactly this roster;
  * do **not** do a teammate's work before that teammate is available — create the task, assign it to the exact roster name, and wait for the owner.
* **The same rules are appended to `buildPersistentLeadContext`**, so launch hydration and post-compact recovery reinforce them rather than only the create path having them. A solo lead (no teammates) gets no roster-rules block.
* **Degenerate cases pass through untouched:** an empty prompt stays empty so create mode still skips writing a deferred prompt file, and a team with no teammates gets the raw prompt.
* The roster formatting helpers (`indentMultiline`, `formatWorkflowBlock`, `buildMembersPrompt`, `buildCompactMembersRoster`) move out of `TeamProvisioningPromptBuilders.ts` into a focused `TeamProvisioningRosterPrompt.ts`, joined by `getTeammateRosterMembers` and `buildLeadRosterIntegrityRules`. Both callers now build the roster through one module instead of two copies drifting apart, and `TeamProvisioningPromptBuilders.ts` drops below its frozen size cap in the process.

This is prompt-level guidance, not an enforcement mechanism — it does not stop a model from misbehaving, it removes the information gap that was causing it.

## Tests

New `src/main/services/team/provisioning/__tests__/TeamProvisioningRosterPrompt.test.ts` (9 tests):

* `getTeammateRosterMembers` filters lead-role entries out of the delegable roster.
* `buildLeadRosterIntegrityRules` names the exact roster and contains the "never invent" and "wait for the owner" rules.
* `buildCreateBootstrapUserPrompt` wraps the deferred prompt with the roster and rules; excludes the lead from both the roster and the EXACTLY list; passes the prompt through untouched when there are no teammates; stays empty for an empty prompt so create mode still skips the deferred prompt file.
* `buildDeterministicLaunchHydrationPrompt` / `buildPersistentLeadContext` carry the exact teammate names and the integrity rules, keep them in the compact post-compact context, and omit the roster-rules block for a solo lead.

Existing suites re-run green: `TeamProvisioningCreateDeterministicSpawnFlow.test.ts` (16), `TeamProvisioningCreateDeterministicRunFlow.test.ts` (4), `TeamProvisioningIdlePromptInjection.test.ts` (5), `TeamProvisioningPromptBuilders.test.ts` (5), `TeamProvisioningLaunchFailurePolicy.test.ts` (6) — 45 tests total.

## Tested on

Windows 11 (Electron dev build).

* Reproduced and verified with **mixed teams**: a **Cursor/opencode** lead with **local models via opencode** teammates (the configuration where the fabricated-name behavior was most reliable), and with an **Anthropic (API key)** lead. Ran the exact "one task per teammate" prompt before and after: before, invented owners and lead-authored files; after, one task per real roster name with the lead waiting for bootstrap.
* Also smoke-checked create mode with **Codex (ChatGPT)** and **Anthropic (OAuth)** leads to confirm the wrapped prompt does not disturb normal single-instruction runs.
* Not exercised: **Gemini**, macOS and Linux. The change is prompt text plus roster filtering, with no provider-specific branching.

## References

* #113 — lead invents teammate names and does teammates' work on the first create-mode prompt
* #439 — partially: the lead acting before teammates are available

## Validation checklist

- [x] `pnpm typecheck` — exit 0
- [x] Targeted vitest files — 45 tests passing
- [x] `pnpm guard:source-file-size` — passes (`TeamProvisioningPromptBuilders.ts` drops 1160 -> 1130, under its frozen cap; the new module is 91 lines; the baseline file is deliberately left untouched so this PR does not conflict with the others in this batch)
- [x] `pnpm guard:team-provisioning-architecture` — passes
- [x] `pnpm exec eslint` on touched files — 0 errors
- [x] `pnpm exec prettier --check` on touched files — clean

---
**Authorship & provenance:** All code in this PR was written by **Claude (Fable 5)**, directed and live-verified on Windows by me. As with anything under AGPL-3.0, it is offered **as-is, without warranty of any kind** - use, adapt, or reject freely. Full context: Discord thread, Aug 30. Review questions are welcome; answers will come from the same Claude-assisted workflow.

## Summary by CodeRabbit

* **New Features**
  * Improved team setup prompts with clearer teammate rosters, roles, models, and workflows.
  * Added safeguards to prevent invented teammate names and duplicated work during team startup.
  * Preserved the original request while providing richer context during setup and ongoing interactions.
  * Improved handling of lead and teammate roles for more accurate roster assignments.

* **Bug Fixes**
  * Improved roster consistency across initial setup and persistent team context.

* **Tests**
  * Added coverage for roster filtering, prompt generation, edge cases, and solo-team behavior.

### CODE REVIEW

File: src/main/services/team/provisioning/TeamProvisioningPromptBuilders.ts [hunk 1]
```diff
@@ -20,6 +20,14 @@ import { buildActionModeProtocol } from '../actionModeInstructions';
...
 
 import { getAgentLanguageInstruction } from './TeamProvisioningAgentLanguage';
+import {
+  buildCompactMembersRoster,
+  buildLeadRosterIntegrityRules,
+  buildMembersPrompt,
+  formatWorkflowBlock,
+  getTeammateRosterMembers,
+  indentMultiline,
+} from './TeamProvisioningRosterPrompt';
```
[the remark this answers is not in the document]
Comment (NeatCode-Labs) [reply src/main/services/team/provisioning/TeamProvisioningPromptBuilders.ts:30]: Good catch on the module contract — `buildMembersPrompt` and `buildCompactMembersRoster` were indeed exported from `TeamProvisioningPromptBuilders.ts` on `dev` before this PR. There was no compile break (no caller outside the provisioning folder imported them, so `pnpm typecheck` passed either way), but the public surface was narrowed unnecessarily. Fixed in `67978be` by re-exporting both from `TeamProvisioningPromptBuilders.ts`; typecheck is clean and the roster prompt tests (9) still pass.

File: src/main/services/team/provisioning/TeamProvisioningRosterPrompt.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+function isLeadRosterMember(member: TeamCreateRequest['members'][number]): boolean {
+  return Boolean(member.role?.toLowerCase().includes('lead'));
+}
+
+/** Roster entries the lead may delegate to — every configured member except the lead itself. */
+export function getTeammateRosterMembers(
+  members: TeamCreateRequest['members']
+): TeamCreateRequest['members'] {
+  return members.filter((member) => !isLeadRosterMember(member));
```
[the remark this answers is not in the document]
Comment (NeatCode-Labs) [reply src/main/services/team/provisioning/TeamProvisioningRosterPrompt.ts:62]: Good catch — valid. `role` there is free-form teammate text, so `includes('lead')` dropped legitimate teammates like "Frontend lead" or "team leader" from the roster and the EXACTLY list, which is exactly the failure this prompt is meant to prevent. It also never did its stated job: create-mode members never contain the lead (BootstrapSpec models it under `spec.lead`) and launch-mode expected members already exclude it via `isLaunchLeadMember`. Fixed by reusing `isLeadMember` + `isReservedLeadRole` from `@shared/utils/leadDetection`, with tests covering "Frontend lead" staying and the lead still being excluded. (Commit d6fa57fc3.)

File: src/main/services/team/provisioning/TeamProvisioningPromptBuilders.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (NeatCode-Labs) [reply src/main/services/team/provisioning/TeamProvisioningPromptBuilders.ts]: Good catch on the module contract — `buildMembersPrompt` and `buildCompactMembersRoster` were indeed exported from `TeamProvisioningPromptBuilders.ts` on `dev` before this PR. There was no compile break (no caller outside the provisioning folder imported them, so `pnpm typecheck` passed either way), but the public surface was narrowed unnecessarily. Fixed in `67978be` by re-exporting both from `TeamProvisioningPromptBuilders.ts`; typecheck is clean and the roster prompt tests (9) still pass.
